spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonApplicationContextInitializer.java

```diff
@@ -0,0 +1,38 @@
+package org.springframework.cloud.netflix.ribbon;
+
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+
+import java.util.List;
+
+/**
+ * Responsible for eagerly creating the child application context holding the Ribbon
+ * related configuration
+ *
+ * @author Biju Kunjummen
+ */
+public class RibbonApplicationContextInitializer
```

Comment:
I have introduced a configuration properties now @spencergibb - `RibbonEagerLoadProperties` with the following properties:

`ribbon.eager-load.enabled`, `ribbon.eager-load.serviceIds` and driving the eager registration of Ribbon Client configurations based on these